test/unit/botocore/test_awsrequest.py

```diff
@@ -0,0 +1,348 @@
+            )
+            conn = AWSHTTPConnection('s3.amazonaws.com', 443)
+            conn.sock = s
+            wait_mock.return_value = True
+            conn.request('GET', '/bucket/foo', b'body', {'Expect': b'100-continue'})
+            # Assert that we waited for the 100-continue response
+            self.assertEqual(wait_mock.call_count, 1)
+            response = conn.getresponse()
+            # Now we should verify that our final response is the 307.
+            self.assertEqual(response.status, 307)
+
+    def test_expect_100_continue_no_response_from_server(self):
+        with mock.patch('urllib3.util.wait_for_read') as wait_mock:
+            # Shows the server first sending a 100 continue response
+            # then a 200 ok response.
+            s = FakeSocket(
+                b'HTTP/1.1 307 Temporary Redirect\r\n'
+                b'Location: http://example.org\r\n'
+            )
+            conn = AWSHTTPConnection('s3.amazonaws.com', 443)
+            conn.sock = s
+            # By settings wait_mock to return False, this indicates
+            # that the server did not send any response.  In this situation
+            # we should just send the request anyways.
+            wait_mock.return_value = False
+            conn.request('GET', '/bucket/foo', b'body', {'Expect': b'100-continue'})
+            # Assert that we waited for the 100-continue response
+            self.assertEqual(wait_mock.call_count, 1)
+            response = conn.getresponse()
+            self.assertEqual(response.status, 307)
+
+    def test_message_body_is_file_like_object(self):
+        # Shows the server first sending a 100 continue response
+        # then a 200 ok response.
+        body = BytesIOWithLen(b'body contents')
+        s = FakeSocket(b'HTTP/1.1 200 OK\r\n')
+        conn = AWSHTTPConnection('s3.amazonaws.com', 443)
+        conn.sock = s
+        conn.request('GET', '/bucket/foo', body)
+        response = conn.getresponse()
+        self.assertEqual(response.status, 200)
+
+    def test_no_expect_header_set(self):
+        # Shows the server first sending a 100 continue response
+        # then a 200 ok response.
+        s = FakeSocket(b'HTTP/1.1 200 OK\r\n')
+        conn = AWSHTTPConnection('s3.amazonaws.com', 443)
+        conn.sock = s
+        conn.request('GET', '/bucket/foo', b'body')
+        response = conn.getresponse()
+        self.assertEqual(response.status, 200)
+
+    def test_tunnel_readline_none_bugfix(self):
+        # Tests whether ``_tunnel`` function is able to work around the
+        # py26 bug of avoiding infinite while loop if nothing is returned.
+        conn = self.create_tunneled_connection(
+            url='s3.amazonaws.com',
+            port=443,
+            response=b'HTTP/1.1 200 OK\r\n',
+        )
+        conn._tunnel()
+        # Ensure proper amount of readline calls were made.
+        self.assertEqual(self.mock_response.fp.readline.call_count, 2)
```

Comment:
lets throw out test that are not useful to us so we don't waste time on maintenence